src/poetry/mixology/version_solver.py

```diff
@@ -47,13 +46,7 @@ def __init__(self, provider: Provider) -> None:
             list[DependencyPackage],
         ] = {}
 
-        # TODO: re-enable cache when poetry-core upgrade is completed
-        self.search_for = functools.lru_cache(
-            maxsize=128
-            if metadata.version("poetry-core")  # type: ignore[no-untyped-call]
-            != "1.1.0a7"
-            else 0
-        )(self._search_for)
+        self.search_for = functools.lru_cache(maxsize=128)(self._search_for)
```

Comment:
Can we use the decorator syntax again? 😄

Reply:
nope, this causes memory leaks

Reply:
also `flake8-bugbear` will point it out as memory leak

Reply:
What @Secrus said, this was actually a fix for memory leak :D we could have done the conditional on decorator as well if we wanted.

Reply:
Yep, since https://github.com/python-poetry/poetry/pull/5662, using a decorator on class instances methods is reported by `flake8-bugbear` as a potential bad thing to do, as this prevents objects from being garbage collected correctly.

The rationale is explained in those 2 resources:
- https://rednafi.github.io/reflections/dont-wrap-instance-methods-with-functoolslru_cache-decorator-in-python.html
- https://www.youtube.com/watch?v=sVjtp6tGo0g

PR on `flake8-bugbear` side, for even more background: https://github.com/PyCQA/flake8-bugbear/pull/218.